stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+  plt.xlabel('Predykcja')
+  plt.ylabel('Rzeczywistość')
+  plt.show()
+
+def print_classification_report(model, X: pd.DataFrame, y: pd.DataFrame) -> None:
+  y_predict = model.predict(X)
+  print(classification_report(y, y_predict))
+
+def show_metrics(model, X: pd.DataFrame, y: pd.DataFrame) -> None:
+  y_pred = model.predict(X)
+#  print("Macierz błędu\n", confusion_matrix(y_true, y_pred), "\n")
+  accuracy = accuracy_score(y, y_pred)
+  print(f"Dokładność (accuracy): {accuracy*100:.2f}%")
+  precision = precision_score(y, y_pred)
+  print(f"Precyzja (precision): {precision*100:.2f}%")
+  recall = recall_score(y, y_pred)
+  print(f"Czułość (recall): {recall*100:.2f}%")
+  f1 = f1_score(y, y_pred)
+  print(f"F1-score: {f1*100:.2f}%")
+
+def plot_map(df: pd.DataFrame, parameter_name: str, colormap: mpl.colors.LinearSegmentedColormap,
+             title: str, point_size: int = 8, width: int = 900, height: int = 600, alpha: float = 1,
+             bgcolor: str = 'white'):
+
+    gdf = gv.Points(df, ['lon', 'lat'], [parameter_name]) # obiekt zawierający punkty
+    tiles = gts.OSM # wybór mapy tła, w tym wypadku OpenStreetMap
+
+    # łączenie mapy tła z punktami i ustawienie wybranych parametrów wizualizacji
+    map_with_points = tiles * gdf.opts(
+        title=title,
+        color=parameter_name,
+        cmap=colormap,
+        size=point_size,
+        width=width,
+        height=height,
+        colorbar=True,
+        toolbar='above',
+        tools=['hover', 'wheel_zoom', 'reset'],
+        alpha=alpha, # przezroczystość
+        bgcolor=bgcolor
+    )
+
+    map_with_points.opts(bgcolor=bgcolor)
+
+#    # zapis mapy do pliku .html
+#    output_filename = f'/content/drive/MyDrive/BigMess/NASA/output_map_{parameter_name}.html'
+#    hv.save(map_with_points, output_filename)
+
+    return hv.render(map_with_points)
+
+"""# Las losowy
+
+## Trening i test modelu przy użyciu nowych danych i zbioru anotowanego
+
+### Średnie 3-letnie
+"""
+
+X = monthly_avg_3_july2023_an_merge.drop(columns=['lon', 'lat', 'pustynia'])
+y = monthly_avg_3_july2023_an_merge['pustynia']
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=2023)
+
+rf_monthly_avg_3_july2023 = RandomForestClassifier(max_depth=8, min_samples_leaf=4, n_estimators=70, random_state=3)
+
+rf_monthly_avg_3_july2023.fit(X_train, y_train)
+
+"""Zastosujmy model na pełnych danych:"""
+
+monthly_avg_3_july2023_pred = monthly_avg_3_july2023.copy()
+monthly_avg_3_july2023_pred['pustynia'] = rf_monthly_avg_3_july2023.predict(monthly_avg_3_july2023.loc[:, [col for col in monthly_avg_3_july2023.columns if col.startswith('monthly_avg')]])
+
+output_notebook()
+show(plot_map(df=monthly_avg_3_july2023_pred, parameter_name='pustynia',
+              colormap=dict(zip(['1', '0'], ['yellow', 'green'])),
+              title='Pustynie (1) i niepustynie (0) - las losowy (lipiec 2023, średnia 3-letnia)',
+              point_size=3, alpha=0.7))
```

Comment:
Po wywołaniu tego kodu, w notatniku widzimy takie powiadomienie 
![screen](https://github.com/PiotrMaciejKowalski/BigData2024Project/assets/148125431/44165bf3-441a-4b70-9cd7-412df3877295) które może warto spróbować usunąć. To samo powiadomienie widzimy za każdym razem jak tworzysz mapę.

Reply:
Warning jest rzucany przez naszą zależność. Można by wyciszyć warningi - ale to chyba poradzimy sobie bez. Ja bym tu nic nie robił.